mqtt/mqtt-bridge/src/bridge.rs

```diff
@@ -34,72 +42,94 @@ impl BridgeShutdownHandle {
 
 /// Bridge implementation that connects to local broker and remote broker and handles messages flow
 pub struct Bridge {
-    pumps: PumpPair,
+    local_pump: Pump<LocalUpstreamHandler<WakingMemoryStore>, LocalUpstreamPumpEventHandler>,
+    remote_pump: Pump<RemoteUpstreamHandler<WakingMemoryStore>, RemoteUpstreamPumpEventHandler>,
     connection_settings: ConnectionSettings,
 }
 
 impl Bridge {
     pub async fn new(
         system_address: String,
         device_id: String,
-        connection_settings: ConnectionSettings,
+        settings: ConnectionSettings,
     ) -> Result<Self, BridgeError> {
-        debug!("creating bridge {}...", connection_settings.name());
-
-        let mut pumps = PumpPair::new(&connection_settings, &system_address, &device_id)?;
-
-        pumps
-            .local_pump
+        const BATCH_SIZE: usize = 10;
+
+        debug!("creating bridge...");
+
+        let (mut local_pump, mut remote_pump) = Builder::default()
+            .with_local(|pump| {
+                pump.with_config(MqttClientConfig::new(
+                    &system_address,
+                    settings.keep_alive(),
+                    settings.clean_session(),
+                    Credentials::Anonymous(format!(
+                        "{}/$edgeHub/$bridge/{}",
+                        device_id,
+                        settings.name()
+                    )),
+                ))
+                .with_rules(settings.forwards());
+            })
+            .with_remote(|pump| {
+                pump.with_config(MqttClientConfig::new(
+                    settings.address(),
+                    settings.keep_alive(),
+                    settings.clean_session(),
+                    settings.credentials().clone(),
+                ))
+                .with_rules(settings.subscriptions());
+            })
+            .with_store(|| PublicationStore::new_memory(BATCH_SIZE))
+            .build()?;
+
+        // TODO move susbscriptions into run method
```

Comment:
What is the intention of this change? Without some change to the bridge controller, this will have the same behavior.

Reply:
the pump subscriptions can be changed with the twin update. so it has to be changed to react to changes. we discussed with @ancaantochi how we can achieve this with changes in her PR